module/VuFindTheme/src/VuFindTheme/ResourceContainer.php

```diff
@@ -130,13 +130,100 @@ public function addCss($css)
+        } elseif (isset($js[0])) {
+            foreach ($js as $current) {
+                $this->addJsEntry($current);
+            }
+        } else {
+            trigger_error("Invalid JS entry format: " . print_r($js, true));
+        }
+    }
+
+    /**
+     * Helper function for adding a Javascript file.
+     *
+     * @param string|array $jsEntry Entry to add, either as string with path
+     * or array with additional properties.
+     *
+     * @return void
+     */
+    protected function addJsEntry($jsEntry)
+    {
+        if (!is_array($jsEntry)) {
+            $parts = $this->parseSetting($jsEntry);
+            if (count($parts) == 1) {
+                $jsEntry = ['file' => $jsEntry];
+            } else {
+                $jsEntry = [
+                    'file' => $parts[0],
+                    'attributes' => ['conditional' => trim($parts[1])],
+                ];
+            }
         }
-        foreach ($js as $current) {
-            $this->js[] = $current;
+        if (!isset($jsEntry['position'])) {
+            $jsEntry['position'] = 'header';
+        }
+
+        if (!in_array($jsEntry, $this->js, true)) {
+            $this->insertEntry($jsEntry, $this->js);
         }
     }
 
+    /**
+     * Helper function to insert an entry to an array,
+     * also considering prioroty and dependancy, if existing.
+     *
+     * @param array $entry The entry to insert.
+     * @param array $array The array into which the entry shall be inserted.
+     *
+     * @return void
+     */
+    protected function insertEntry($entry, &$array)
+    {
+        if (isset($entry['priority']) || isset($entry['dependency'])) {
+            for ($i=0; $i < count($array);++$i) {
+                if (isset($entry['priority'])) {
+                    $currentPriority = $array[$i]['priority'] ?? null;
+                    if (!isset($currentPriority)
+                        || $currentPriority > $entry['priority']
+                    ) {
+                        $this->insertEntryAtPosition($entry, $i, $array);
+                        return;
+                    }
+                } elseif (isset($entry['dependency'])) {
+                    if ($entry['dependency'] == $array[$i]['file']) {
+                        $this->insertEntryAtPosition($entry, $i + 1, $array);
+                        return;
+                    }
+                }
+            }
+        }
+
+        // Insert at end if either no priority/dependency is given
+        // or no other element has been found
+        $array[] = $entry;
+    }
+
+    /**
+     * Helper function to insert an element into an array
+     * at a certain position.
+     *
+     * @param array $entry    The entry to be inserted.
+     * @param int   $position The position to insert the entry.
+     * @param array $array    The array in which the entry shall be inserted.
+     *
+     * @return void
+     */
+    protected function insertEntryAtPosition($entry, $position, &$array)
```

Comment:
Do you need this support method? I think `array_splice($array, $position, 0, $entry)` will do the same thing with native PHP.

Reply:
You're right! Due to the documentation, i didn't think it would be allowed to pass 0 as length, but the test case seems to work.

Reply:
Great!